### ISSUE TITLE
Incompatible with unpublished thrid-party reporter

### ISSUE THREAD

**Issue by ouchxp** [zinserjan/mocha-webpack#58]
I have a third-party mocha reporter would like to use, but `checkReporter` does not allow that.
In `checkReporter` we have two checks, `require("mocha/lib/reporters/" + reporter);` and `require(reporter);` which applies to mocha's built-in reporter and published reporter. However unpublished reporter in my project will always fail.

```
Error: reporter "test/test-lib/specReporterWithLineNumber" does not exist
...
```

I think an extra `require("mocha/../../" + reporter);` to check against unpublished thrid-party reporter would be nice to have. It's a bad pattern though. Currently I have to use `NODE_PATH=.` to solve this issue, which is not ideal either.

**Comment by zinserjan** [comment]
Mocha doesn't support this at the moment.

**Comment by ouchxp** [comment]
@zinserjan What is mocha not supporting? I'm currently using third-party reporter I wrote, and it works fine. I put the reporter at `test/test-lib/specReporterWithLineNumber.js` and executing it with script like this `NODE_PATH=test/test-lib mocha-webpack --webpack-config webpack.config.test.js --reporter specReporterWithLineNumber ...`

**Comment by zinserjan** [comment]
Fixed by #87.

`Node_PATH` hack is no longer necessary. The following works:
```
mocha-webpack --webpack-config webpack.config.test.js --reporter test/test-lib/specReporterWithLineNumber.js ...
```
